test-interface/src/main/scala/scala/scalanative/testinterface/NativeRPC.scala

```diff
@@ -21,28 +21,23 @@ private[testinterface] class NativeRPC(clientSocket: Socket) extends RPCCore {
 
   @tailrec
   private[testinterface] final def loop(): Int = {
-    def tryRead: Try[Boolean] = Try {
-      val msgLength = inStream.readInt()
-
-      /**
-       * Current implementation of DataInputStream does not check for EOF,
-       * in this case we need to follow up base `read` behaviour which is returning -1 value to signal EOF
-       * TODO Fix this after merging changes due to #1868
-       */
-      if (msgLength < 0) true
-      else {
-        val msg = Array.fill(msgLength)(inStream.readChar).mkString
-        handleMessage(msg)
-        scalanative.runtime.loop()
-        false
+    val msgLength =
+      try {
+        inStream.readInt()
+      } catch {
+        case _: EOFException => 0 // leave loop
+        case throwable: Throwable =>
```

Comment:
Would it be possible to throw anything other then `EOFException` in new implementation?

Reply:
Thank you for looking a this and your discussion/suggestions.

I have been taught that I/O can always throw some evil, and to be
prepared.